src/gluonnlp/data/transforms.py

```diff
@@ -988,10 +1004,261 @@ def convert_tokens_to_ids(self, tokens):
+            # Account for [CLS], [SEP], [SEP] with "- 3"
+            _truncate_seq_pair(tokens_a, tokens_b, self._max_seq_length - 3)
+        else:
+            # Account for [CLS] and [SEP] with "- 2"
+            if len(tokens_a) > self._max_seq_length - 2:
+                tokens_a = tokens_a[0:(self._max_seq_length - 2)]
+
+        # The embedding vectors for `type=0` and `type=1` were learned during
+        # pre-training and are added to the wordpiece embedding vector
+        # (and position vector). This is not *strictly* necessary since
+        # the [SEP] token unambiguously separates the sequences, but it makes
+        # it easier for the model to learn the concept of sequences.
+
+        # For classification tasks, the first vector (corresponding to [CLS]) is
+        # used as as the "sentence vector". Note that this only makes sense because
+        # the entire model is fine-tuned.
+        tokens = []
+        segment_ids = []
+        tokens.append('[CLS]')
+        segment_ids.append(0)
+        for token in tokens_a:
+            tokens.append(token)
+            segment_ids.append(0)
+        tokens.append('[SEP]')
+        segment_ids.append(0)
+
+        if tokens_b:
+            for token in tokens_b:
+                tokens.append(token)
+                segment_ids.append(1)
+            tokens.append('[SEP]')
+            segment_ids.append(1)
+
+        input_ids = self._tokenizer.convert_tokens_to_ids(tokens)
+
+        # The valid length of sentences. Only real  tokens are attended to.
+        valid_length = len(input_ids)
+
+        if self._pad:
+            # Zero-pad up to the sequence length.
+            padding_length = self._max_seq_length - valid_length
+            # use padding tokens for the rest
+            input_ids.extend([self._tokenizer.vocab['[PAD]']] * padding_length)
+            segment_ids.extend([self._tokenizer.vocab['[PAD]']] * padding_length)
+
+        return np.array(input_ids, dtype='int32'), np.array(valid_length, dtype='int32'),\
+            np.array(segment_ids, dtype='int32')
+
+
+class BERTClassificationTransform(object):
+    """Dataset Transformation for BERT-style Sentence Classification.
+
+    Parameters
+    ----------
+    tokenizer : BERTTokenizer.
+        Tokenizer for the sentences.
+    labels : list of int.
+        List of all label ids for the classification task.
+    max_seq_length : int.
+        Maximum sequence length of the sentences.
+    pad : bool, default True
+        Whether to pad the sentences to maximum length.
+    pair : bool, default True
+        Whether to transform sentences or sentence pairs.
+    """
+
+    def __init__(self, tokenizer, labels, max_seq_length, pad=True, pair=True):
+        self._label_map = {}
+        for (i, label) in enumerate(labels):
+            self._label_map[label] = i
+        self._bert_xform = BERTTransform(tokenizer, max_seq_length, pad=pad, pair=pair)
+
+    def __call__(self, line):
```

Comment:
Do you mind also adding `__call__` to "special members" so that the doc for `__call__` would display? https://github.com/dmlc/gluon-nlp/blob/master/docs/api/modules/data.rst#api-reference

Reply:
I will add this after the API has met the requirements.